### PR TITLE
Add key support

### PR DESCRIPTION
**Pull request by vrde**
Quick and dirty implementation of key submission and display. Not an ERC1155.

Since keys are from Alice, Bob, ..., instead of a generic :key: we can also display the initial, so 🇦, 🇧, ...

https://user-images.githubusercontent.com/134680/194748089-22d629ea-a702-4709-b850-57bb9cfcfe35.mp4

### CODE REVIEW

File: eth/test/integration.ts [hunk 1]
```diff
@@ -150,14 +160,34 @@ describe("TreasureHuntCreator", () => {
...
       expect(result).equal(deployer.address);
     });
   });
 
+  describe("submitKey", async () => {
+    it("should add a key to the player bitmap", async () => {
+      const instance = await deploy(solutions, keys, questsRootCid);
+      const { r, v, s } = await getKeySignature(MNEMONICS[0], deployer.address);
+      await instance.connect(deployer).submitKey(v, r, s);
+      let result = await instance._playerToKeys(deployer.address);
```
Comment (sirnicolaz) [inline eth/test/integration.ts:176]: there is no assertion here. is it expected?

File: eth/test/test_pagination.ts [hunk 2]
```diff
@@ -32,14 +36,16 @@ describe("TreasureHuntCreator", () => {
...
+      "QmUYWv6RaHHWkk5BMHJH4xKPEKNqAYKomeiTVobAMyxsbz"
+    );
   });
 
   async function deploy(
     solutions: string[],
     questsRootCid: Uint8Array
   ): Promise<TreasureHuntCreator> {
-    const thc = await thcFactory.deploy(solutions, questsRootCid);
+    const thc = await thcFactory.deploy(solutions, [], questsRootCid);
```
Comment (sirnicolaz) [inline eth/test/test_pagination.ts:48]: shouldn't there be a test that checks that the leaderboard result contains the keys as expected?

File: eth/test/integration.ts (hunk 1 above)
Comment (vrde) [reply eth/test/integration.ts:176]: No, it's a mistake, good catch.

File: eth/test/test_pagination.ts (hunk 2 above)
Comment (vrde) [reply eth/test/test_pagination.ts:48]: This test only checks the pagination is correct. The test you are talking about is implemented in the `integration.ts` test suite, specifically here https://github.com/social-dist0rtion-protocol/thc/blob/be6dc51f93ed0d88b6b6bc3ca1ad9fc70fece79d/eth/test/integration.ts#L279-L285

**Review by sirnicolaz — APPROVED** [review]

**Review by sirnicolaz — APPROVED** [review]

**Pull request merged**